R/new-model.R

```diff
@@ -28,51 +29,48 @@
 #'   `.yaml_path`. If `TRUE` overwrite existing file, if one exists.
 #' @param .model_type Character scaler to specify type of model being created
 #'   (used for S3 class). Currently only `'nonmem'` is supported.
-#' @param .directory Model directory which `.yaml_path` is relative to. Defaults
-#'   to `options('rbabylon.model_directory')`, which can be set globally with
-#'   `set_model_directory()`.
-#' @importFrom yaml write_yaml
-#' @importFrom fs file_exists
-#' @importFrom digest digest
+#'
 #' @return S3 object of class `bbi_{.model_type}_model` that can be passed to
 #'   `submit_model()`, `model_summary()`, etc.
 #' @export
 new_model <- function(
-  .yaml_path,
+  .path,
   .description,
   .based_on = NULL,
   .tags = NULL,
   .bbi_args = NULL,
   .overwrite = FALSE,
-  .model_type = c("nonmem"),
-  .directory = get_model_directory()
+  .model_type = c("nonmem")
 ) {
 
-  # TODO: update once numeric "dispatch" is formally deprecated
-  if (fs::path_ext(as.character(.yaml_path)) != "yaml") {
-      warning(glue("Did not pass a YAML extension to .yaml_path. Inferred path `{yaml_ext(.yaml_path)}` from `{.yaml_path}`"))
-      .yaml_path <- yaml_ext(.yaml_path)
-  }
-
-  # check for .directory and combine with .yaml_path
-  .yaml_path <- combine_directory_path(.directory, .yaml_path)
-
+  maybe_yaml_path <- paste0(.path, ".yaml")
   # check if file already exists
-  if (fs::file_exists(.yaml_path) && !isTRUE(.overwrite)) {
-    stop(paste(glue("Passed {.yaml_path} to `new_model(.yaml_path)` but that file already exists."),
-               "Either call `read_model()` to load model from YAML or use `new_model(.overwrite = TRUE)` to overwrite the existing YAML."))
+  if (fs::file_exists(maybe_yaml_path) && !isTRUE(.overwrite)) {
+    stop(
+      glue::glue(
+        "File already exists at {maybe_yaml_path}.",
+        "Either call `read_model()` to load model from YAML or use,",
+        "`new_model(.overwrite = TRUE)` to overwrite the existing YAML.",
+        .sep = " "
+      )
+    )
   }
 
-  model_working_directory <- normalizePath(dirname(.yaml_path))
-  model_id <- get_model_id(.yaml_path)
+  # construct the absolute model path in a way that avoids a warning from
+  # normalizePath() if `.path` does not exist (we only require that the model
+  # file exist at `.path`)
+  abs_mod_path <- file.path(
+    normalizePath(dirname(.path)),
```

Comment:
should this have `mustWork = TRUE`? Am I right that the _directory_ we are writing into needs to exist? If so, might as well catch it here.

Reply:
What do you mean by 

> the _directory_ we are writing into

If the model file is at `/path/to/foo/bar.ctl`, we require only that `bar.ctl` exists in `/path/to/foo`. This check is performed by `find_nonmem_model_file_path()`, which we call in `create_model_object()`.

Reply:
ah yes, I forgot that we check for the model file. That will verify that the directory exists.